File: apps/api/internal/model/cluster.go
package model

// EntityCounts represents total and running counts for an entity type
type EntityCounts struct {
	Total   int `json:"total"`
	Running int `json:"running,omitempty"`
}

// CPUAggregate represents aggregated CPU information
type CPUAggregate struct {
	Cores    int     `json:"cores"`
	AvgUsage float64 `json:"avgUsage"`
}

// ClusterSummary provides an overview of the entire cluster
type ClusterSummary struct {
	Nodes      EntityCounts  `json:"nodes"`
	VMs        EntityCounts  `json:"vms"`
	Containers EntityCounts  `json:"containers"`
	Stacks     EntityCounts  `json:"stacks"`
	CPU        CPUAggregate  `json:"cpu"`
	Memory     MemoryInfo    `json:"memory"`
	Disk       DiskInfo      `json:"disk"`
}

// TimeSeriesDataPoint represents a single data point in a time series
type TimeSeriesDataPoint struct {
	Time  string  `json:"time"`
	Value float64 `json:"value"`
}

// ClusterMetrics contains time series data for cluster metrics
type ClusterMetrics struct {
	CPUUsage    []TimeSeriesDataPoint `json:"cpuUsage"`
	MemoryUsage []TimeSeriesDataPoint `json:"memoryUsage"`
	NetworkIn   []TimeSeriesDataPoint `json:"networkIn"`
	NetworkOut  []TimeSeriesDataPoint `json:"networkOut"`
}

File: apps/web/lib/api/queries/cluster.ts
import { useQuery } from "@tanstack/react-query";
import { clusterClient } from "../client";
import type { ClusterSummary } from "../types";

export function useClusterSummary() {
  return useQuery({
    queryKey: ["cluster", "summary"],
    queryFn: () => clusterClient.getClusterSummary({}),
    select: (res): ClusterSummary => ({
      nodes: { total: res.nodes?.total ?? 0, running: res.nodes?.running ?? 0 },
      vms: { total: res.vms?.total ?? 0, running: res.vms?.running ?? 0 },
      containers: {
        total: res.containers?.total ?? 0,
        running: res.containers?.running ?? 0,
      },
      stacks: {
        total: res.stacks?.total ?? 0,
        running: res.stacks?.running ?? 0,
      },
      cpu: { cores: res.cpu?.cores ?? 0, avgUsage: res.cpu?.avgUsage ?? 0 },
      memory: { used: res.memory?.used ?? 0, total: res.memory?.total ?? 0 },
      disk: { used: res.disk?.used ?? 0, total: res.disk?.total ?? 0 },
    }),
  });
}

export function useClusterMetrics(points?: number) {
  return useQuery({
    queryKey: ["cluster", "metrics", points],
    queryFn: () => clusterClient.getClusterMetrics({ points: points ?? 0 }),
  });
}
